.github/workflows/codecov_ats.yml

```diff
@@ -80,13 +87,43 @@ jobs:
           env:
             CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
             CODECOV_STATIC_TOKEN: ${{ secrets.CODECOV_STATIC_TOKEN }}
-        # Run label analysis IN DRY MODE (no tests will actually run)
-        - name: Label Analysis
+        # Run Automated Test Selection in dry mode to get the list of tests to run
+        # The base commit will be the parent commit (apparently commits on master don't exist in codecov)
+        - name: Codecov Automated Test Selection
+          id: codecov_automated_test_selection
           continue-on-error: true
           run: |
-            BASE_COMMIT=$(git merge-base ${{ github.sha }}^ origin/master)
+            BASE_COMMIT=$(git rev-parse ${{ github.sha }}^)
             echo $BASE_COMMIT
-            codecovcli --codecov-yml-path=codecov.yml label-analysis --dry-run --token=${CODECOV_STATIC_TOKEN} --base-sha=${BASE_COMMIT}
+            output=$(codecovcli --codecov-yml-path=codecov.yml label-analysis --dry-run --token=${CODECOV_STATIC_TOKEN} --base-sha=${BASE_COMMIT}) || true
+            if [ -n "${output}" ];
+            then
+              json_output=$(echo $output | grep -o -e "{.*}")
+
+              echo ATS_TESTS_TO_RUN=$(jq <<< $json_output '.runner_options + .ats_tests_to_run | map(@sh) | join(" ")' --raw-output) >> "$GITHUB_OUTPUT"
```

Comment:
Validating via an experiment:

```console
$ json_output='{"runner_options": ["a", "b c"], "ats_tests_to_run": ["d", "e f", "mytest[bar=\"foo\"]"] }'

$ jq <<< "$json_output" -c
{"runner_options":["a","b c"],"ats_tests_to_run":["d","e f","mytest[bar=\"foo\"]"]}

$ GITHUB_OUTPUT=github.out && echo > "$GITHUB_OUTPUT"

$ echo ATS_TESTS_TO_RUN=$(jq <<< $json_output '.runner_options + .ats_tests_to_run | map(@sh) | join(" ")' --raw-output) >> "$GITHUB_OUTPUT"

$ ATS_TESTS_TO_RUN='a' 'b c' 'd' 'e f' 'mytest[bar="foo"]'
b: command not found
exit code: 127

$ cat "$GITHUB_OUTPUT"
ATS_TESTS_TO_RUN='a' 'b c' 'd' 'e f' 'mytest[bar="foo"]'

$ source "$GITHUB_OUTPUT"
b: command not found
exit code: 127
```

It doesn't look like this is going to work.

 I recommend keeping the structure of the structured data until just before it's used:


```console
$ echo ATS_TESTS_TO_RUN=$(jq <<< $json_output '.runner_options + .ats_tests_to_run | @json | @sh' --raw-output) >> "$GITHUB_OUTPUT"

$ cat "$GITHUB_OUTPUT"
ATS_TESTS_TO_RUN='["a","b c","d","e f","mytest[bar=\"foo\"]"]'

$ source $GITHUB_OUTPUT

$ jq <<< "$ATS_TESTS_TO_RUN"
[
  "a",
  "b c",
  "d",
  "e f",
  "mytest[bar=\"foo\"]"
]

$ jq <<< "$ATS_TESTS_TO_RUN" 'join("\u0000")' --raw-output | xargs -r0 pytest
=== test session starts
=== no tests ran in 0.00s
ERROR: found no collectors for ./a
ERROR: found no collectors for ./b c
ERROR: found no collectors for ./d
ERROR: found no collectors for ./e f
ERROR: found no collectors for ./mytest
exit code: 4
```

As a good side-effect, this will handle the situation that the test list is too long for a linux limits.

P1: It's important to get the test names exactly right, and parameterized tests can/do have arbitrary stuff in the test's name.